### PR TITLE
DATAES-445 - Updated scroll API example.

### PR DESCRIPTION
**Pull request by ltsallas**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/asciidoc/reference/elasticsearch-misc.adoc [hunk 1]
```diff
@@ -23,50 +23,50 @@ Page<SampleEntity> sampleEntities =
...
+			return new AggregatedPageImpl<T>((List<T>) result, response.getScrollId());
+		}
+		return new AggregatedPageImpl<T>(Collections.EMPTY_LIST, response.getScrollId());
+	}
+};
+SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(matchAllQuery())
+			.withIndices(INDEX_NAME)
+			.withTypes(TYPE_NAME)
+			.withFields("message")
+			.withQuery(matchAllQuery())
```
Comment (hzpz) [inline src/main/asciidoc/reference/elasticsearch-misc.adoc]: You already set `withQuery` a few lines above.

Comment (ltsallas) [reply src/main/asciidoc/reference/elasticsearch-misc.adoc]: @hzpz Thanks for your comment. I deleted it

File: src/main/asciidoc/reference/elasticsearch-misc.adoc [hunk 2]
```diff
@@ -23,50 +23,49 @@ Page<SampleEntity> sampleEntities =
...
+			sampleEntity.setMessage(message);
+			result.add(sampleEntity);
+		}
+		if (result.size() > 0) {
+			return new AggregatedPageImpl<T>((List<T>) result, response.getScrollId());
+		}
+		return new AggregatedPageImpl<T>(Collections.EMPTY_LIST, response.getScrollId());
+	}
+};
+SearchQuery searchQuery = new NativeSearchQueryBuilder().withQuery(matchAllQuery())
```
Comment (xhaggi) [inline src/main/asciidoc/reference/elasticsearch-misc.adoc]: Please move `.withQuery` to a new line.

File: src/main/asciidoc/reference/elasticsearch-misc.adoc [hunk 3]
```diff
@@ -23,50 +23,49 @@ Page<SampleEntity> sampleEntities =
...
+			.withIndices(INDEX_NAME)
+			.withTypes(TYPE_NAME)
+			.withFields("message")
+			.withPageable(PageRequest.of(0, 10))
+			.build();
+Page<SampleEntity> scroll = elasticsearchTemplate.startScroll(1000, searchQuery, SampleEntity.class, searchResultMapper);
```
Comment (xhaggi) [inline src/main/asciidoc/reference/elasticsearch-misc.adoc]: Please add a new line before for better readability.

File: src/main/asciidoc/reference/elasticsearch-misc.adoc [hunk 4]
```diff
@@ -23,50 +23,49 @@ Page<SampleEntity> sampleEntities =
...
-    if(page != null) {
-        sampleEntities.addAll(page.getContent());
-        hasRecords = page.hasNextPage();
-    }
-    else{
-        hasRecords = false;
-    }
-    }
-}            
+SearchResultMapper searchResultMapper = new SearchResultMapper() {
```
Comment (xhaggi) [inline src/main/asciidoc/reference/elasticsearch-misc.adoc]: It is not necessary to define a custom result mapper so please remove it from this basic example. There is no need to make it more complicated than it is.

**Comment by xhaggi** [comment]
It would also be nice to have a simple example of using the `ElasticsearchTemplate#stream(SearchQuery, Class<T>)` method.

File: src/main/asciidoc/reference/elasticsearch-misc.adoc (hunk 4 above)
Comment (ltsallas) [reply src/main/asciidoc/reference/elasticsearch-misc.adoc]: Yes I removed it , I agree that there is no point to further complicate the example.

File: src/main/asciidoc/reference/elasticsearch-misc.adoc (hunk 3 above)
Comment (ltsallas) [reply src/main/asciidoc/reference/elasticsearch-misc.adoc]: Done

File: src/main/asciidoc/reference/elasticsearch-misc.adoc (hunk 2 above)
Comment (ltsallas) [reply src/main/asciidoc/reference/elasticsearch-misc.adoc]: Done

**Comment by ltsallas** [comment]
@xhaggi Thanks for your feedback. I added a second example for the stream method in the end.

**Pull request closed without merging**